rule/src/main/java/htw/kbe/maumau/rule/service/RulesServiceImpl.java

```diff
@@ -1,32 +1,53 @@
 package htw.kbe.maumau.rule.service;
 
 import htw.kbe.maumau.card.domain.Card;
+import htw.kbe.maumau.card.domain.Label;
 import htw.kbe.maumau.card.domain.Suit;
-import htw.kbe.maumau.rule.domain.Rule;
 import htw.kbe.maumau.player.domain.Player;
 
 public class RulesServiceImpl implements RulesService {
-    public boolean isJoker(Card card, Rule rule) {
-        return false;
+
+    @Override
+    public boolean isCardValid(Card userCard, Card topCard, Suit userWish) {
+
+        if (topCard.getLabel().equals(Label.JACK)) {
+            if (topCard.getLabel().equals(userCard.getLabel())) {
+                return false;
+            } else {
+                return userCard.getSuit().equals(userWish);
+            }
+        } else {
+            if (topCard.getLabel().equals(userCard.getLabel())) {
+                return true;
+
+            } else return topCard.getSuit().equals(userCard.getSuit());
+        }
     }
```

Comment:
```suggestion
    public boolean isCardValid(Card userCard, Card topCard, Suit userWish) {
        if (Objects.nonNull(userWish)) {
            return userCard.getSuit().equals(userWish);
        }
        if (topCard.getLabel().equals(userCard.getLabel() && topCard.getLabel().equals(Label.JACK))) {
            return false;
        }
        return topCard.getLabel().equals(userCard.getLabel()) || topCard.getSuit().equals(userCard.getSuit());
    }
```
Das wäre mein Gegenvorschlag. Die Bedingungen sind jetzt vereinfacht.